pkgs/development/compilers/ghc/with-packages.nix

```diff
@@ -44,6 +47,7 @@ buildEnv {
       rm -f $out/bin/$prg
       makeWrapper ${ghc}/bin/$prg $out/bin/$prg         \
         --add-flags '"-B$NIX_GHC_LIBDIR"'               \
+        ${optionalString exposeGHC "--add-flags '-package ghc'"} \
```

Comment:
I'm the original author. The `ghc` package is hidden by default, which means that you can't just run ghci and start importing GHC modules. It also screws up things like ghc-mod and flycheck. As someone who works a fair bit with the GHC API this is kind of annoying, so I always run `ghc-pkg expose ghc` to un-hide the ghc package.

The point of this flag is to automate that for people like me. I originally wanted to add the `ghc-pkg expose ghc` call to the `postInstall` hook, but ran into a strange permissions issue with `ghc-pkg`, so I took this approach instead.